bokehjs/src/coffee/models/widgets/multiselect.coffee

```diff
@@ -39,15 +39,25 @@ export class MultiSelectView extends InputWidgetView
       if values[el.attr('value')]
         el.attr('selected', 'selected')
     )
+    # Note that some browser implementations might not reduce
+    # the number of visible options for size <= 3.
     @$el.find('select').attr('size', this.model.size)
 
   change_input: () ->
+    # Haven't checked if :focus selector works for IE <= 7
+    is_focused = @$el.find('select:focus').size()
     value = @$el.find('select').val()
     if value
       @model.value = value
     else
       @model.value = []
     super()
+    # Restore focus back to the <select> afterwards,
+    # so that even if python on_change callback is invoked,
+    # focus remains on <select> and one can seamlessly scroll
+    # up/down.
+    if is_focused
+      @$el.find('select').focus()
```

Comment:
Yeah, I assume there is a better way to do this. One (albeit more minor) problem that continues with this implementation is that when scrolling down a `<select>` with a lot of options, when the focus drops off the `<select>` it scrolls back to the top. When this implementation causes the `<select>` to regain focus, it will have been scrolled so that the selected elements are on the bottom of the viewable options. So the viewable options may change and there's also the distraction of seeing the options and scrollbar flicker.